packages/hidp/hidp/api/views.py

```diff
@@ -54,3 +63,30 @@ def get_object(self):
         if self.kwargs.get(self.lookup_url_kwarg or self.lookup_field) == "me":
             return self.request.user
         raise Http404
+
+
+@method_decorator(sensitive_post_parameters("username", "password"), name="dispatch")
+class LoginView(GenericAPIView):
+    permission_classes = []
+    authentication_classes = []
+    serializer_class = LoginSerializer
+    verification_mailer = EmailVerificationMailer
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        # User is authenticated and is allowed to log in.
+        user = serializer.validated_data["user"]
+
+        # Only log in the user if their email address has been verified.
+        if user.email_verified:
+            hidp_auth.login(request, user)
+            return Response(status=HTTPStatus.OK)
```

Comment:
No content is more correct
```suggestion
            return Response(status=HTTPStatus.NO_CONTENT)
```